### PR TITLE
fix/enhance error handling

### PR DESCRIPTION
**Pull request by ruanxin**
This PR fixed some cases return nil error

### CODE REVIEW

File: controllers/manifest_controller.go [hunk 1]
```diff
@@ -76,7 +76,7 @@ func SetupWithManager(
 	if controllerManagedByManager.Complete(ManifestReconciler(mgr, codec, checkInterval)) != nil {
 		return fmt.Errorf("failed to initialize manifest controller by manager: %w", err)
 	}
-	return nil // never end up here :)
```
Comment (ruanxin) [inline controllers/manifest_controller.go:79]: To reviewer:
a misleading (it should be expected to return nil) and unnecessary comments.

Comment (medmes) [reply controllers/manifest_controller.go:79]: True, it was an unnecessary comment on line 79.

**Review by medmes — APPROVED** [review]

File: controllers/kyma_controller.go [hunk 2]
```diff
@@ -104,15 +104,11 @@ func (r *KymaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 
 	kyma := &v1beta2.Kyma{}
 	if err := r.Get(ctx, req.NamespacedName, kyma); err != nil {
-		if util.IsNotFound(err) {
-			// TODO: revisit after runtime-controller gets upgraded
-			// Related issue: https://github.com/kyma-project/lifecycle-manager/issues/579
+		if !util.IsNotFound(err) {
```
Comment (nesmabadr) [inline controllers/kyma_controller.go:107]: why is this condition negated? (Same for the rest)

**Pull request merged**